### DISCUSSION TITLE
Bootstrap class `.borderless` does not seem respected by Quarto

### DISCUSSION THREAD

**Discussion by vincentarelbundock** [quarto-dev/quarto-cli#8444] [Q&A]
Hi all,

I'm running Quarto 1.4.151 and following the instructions on using Bootstrap classes for tables here: https://quarto.org/docs/authoring/tables.html#using-bootstrap-classes

The instructions specifically highlight `.borderless` as a possible class. I expected the tables to have no lines or rules at all; just the text in the cells. However, Quarto does not seem to respect the borderless class, and the table looks the same as default.

Below I paste a screenshot and a reproducible example which shows that (1) `.borderless` doesn't work, but (2) other Bootstrap classes like `.striped` work.

Any ideas?

```markdown
---
format: html
---

| fruit  | price  |
|--------|--------|
| apple  | 2.05   |
| pear   | 1.37   |
| orange | 3.09   |

: Default class


| fruit  | price  |
|--------|--------|
| apple  | 2.05   |
| pear   | 1.37   |
| orange | 3.09   |

: Borderless table does not work {.borderless}

| fruit  | price  |
|--------|--------|
| apple  | 2.05   |
| pear   | 1.37   |
| orange | 3.09   |

: Striped table works {.striped}
```

![borderless](https://github.com/quarto-dev/quarto-cli/assets/987057/f06103aa-ca28-42e0-8909-efbad512e654)

**Comment by mcanouil** [comment]
That is a very old pre-release. Could you upgrade to the current stable version 1.4.459?

If you want to apply your own table styling, you need to disable Quarto processing, see <https://quarto.org/docs/authoring/tables.html#disabling-quarto-table-processing>.

**Comment by vincentarelbundock** [comment]
Thanks! I really should have updated before posting. It seems to solve the problem, removing lines between rows. However, the lines above the table and below stay, even if I disable quarto processing.

Any clues there?

**Reply by mcanouil** [reply]
I believe it would be way easier to user HTML tables in your case.
Markdown tables is not really designed for heavy customisation.

See <https://quarto.org/docs/authoring/tables.html#html-tables>.


Note that when working with HTML and CSS, the best tool is your browser developer/inspect mode.
It allows you to see what's inside your table.
Of course, you could use your favourite IDE that supports HTML in some regards.

**Reply by vincentarelbundock** [reply]
You are right, of course. But in this case, I crafted a Minimal Working Example using markdown to illustrate a problem that also affects HTML tables. My actual use case is the Quarto website for my R package `tinytable`. As you can see in Inspect Mode, the first table at the top of this page is actually `.table table-borderless`, so in principle there should be no rule at all. Yet we see lines at the top and the bottom. And when I create the same table in a standalone document --- outside of Quarto --- I get a proper borderless table without the header rules: https://vincentarelbundock.github.io/tinytable/vignettes/tutorial.html

**Reply by mcanouil** [reply]
Did you disable Quarto processing?
If not, then Quarto will add its own style for harmonisation.

**Reply by vincentarelbundock** [reply]
Yes, I did. I tried disabling quarto processing in the `<table>`  tag, in the YAML preamble, and using the `:::` syntax. No success.

![borderless](https://github.com/quarto-dev/quarto-cli/assets/987057/fa4c7120-05fb-4bfa-833d-bea46d975150)

**Reply by mcanouil** [reply]
Hum, the borders come from Bootstrap `.table`.
it might be a version specific thing.

**Reply by vincentarelbundock** [reply]
Alright, well thanks a lot for taking your time to engage. I guess I'll chalk this up to some maybe version related mystery. Not really the end of the world, but it still bugs me that the output is different in Quarto and standalone files. ¯\\\_(ツ)\_/¯

**Reply by mcanouil** [reply]
Disabling Quarto features to put Bootstrap back gives the expected behaviour.

````qmd
---
format:
  html:
    minimal: true
include-in-header:
  text: |
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.2/dist/css/bootstrap.min.css">
---

```{=html}
<table class="table-borderless table" data-quarto-disable-processing="true">
<thead>
<tr class="header">
<th>fruit</th>
<th>price</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>apple</td>
<td>2.05</td>
</tr>
<tr class="even">
<td>pear</td>
<td>1.37</td>
</tr>
<tr class="odd">
<td>orange</td>
<td>3.09</td>
</tr>
</tbody>
</table>
```
````

**Reply by vincentarelbundock** [reply]
Oh, this is very interesting! Does this suggest that the `data-quarto-disable-processing` option is not doing its job properly?

**Reply by mcanouil** [reply]
Maybe.

The following also works:

````qmd
---
format: html
include-in-header:
  text: |
    <style type="text/css">
    .table-borderless {
      border: none;
    }
    </style>
---

```{=html}
<table class="table-borderless table" data-quarto-disable-processing="true">
<thead>
<tr class="header">
<th>fruit</th>
<th>price</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>apple</td>
<td>2.05</td>
</tr>
<tr class="even">
<td>pear</td>
<td>1.37</td>
</tr>
<tr class="odd">
<td>orange</td>
<td>3.09</td>
</tr>
</tbody>
</table>
```
````

It appears, there is some CSS conflicts somewhere, but I can't figure it out right now.

Current Bootstrap in Quarto 1.4 is v5.3.1 (latest is v5.3.2), so it's something else.

cc @cscheid in case you know what this is.

**Reply by vincentarelbundock** [reply]
FYI, the problem appears to extend to other bootstrap classes, like `table-warning` and friends.

**Reply by mcanouil** [reply]
I am starting to think those classes aren't even part of the distribution.
For instance, the class you get seems to be from:
- https://github.com/quarto-dev/quarto-cli/blob/1d720e8cba2bbf6b47a7f46714babc8c0f7685b0/src/resources/formats/html/bootstrap/dist/scss/_tables.scss#L96-L106
